s3fs/core.py

```diff
@@ -18,10 +17,14 @@
 
 logging.getLogger('boto3').setLevel(logging.WARNING)
 logging.getLogger('botocore').setLevel(logging.WARNING)
-S3_RETRYABLE_ERRORS = (
-    socket.timeout,
-    trans.ReadTimeoutError, trans.IncompleteReadError
-)
+
+try:
+    from boto3.s3.transfer import S3_RETRYABLE_ERRORS
+except ImportError:
+    S3_RETRYABLE_ERRORS = (
+        socket.timeout,
```

Comment:
Agreed.  For the time being I suggest that we merge this and see what exceptions arise that we need to add here.  

It could also be that boto3 has gotten better about retreis.